### PR TITLE
Support custom hashing functions

### PR DESCRIPTION
**Pull request by ernestognw**
## Description

Add support for custom hashing functions.

### CODE REVIEW

File: src/standard.ts [hunk 1]
```diff
@@ -23,58 +23,71 @@ export class StandardMerkleTree<T extends any[]> {
...
+ 
+  static load<T extends any[]>(data: StandardMerkleTreeData<T>, hashFn?: HashFn): StandardMerkleTree<T> {
     if (data.format !== 'standard-v1') {
       throw new Error(`Unknown format '${data.format}'`);
     }
     return new StandardMerkleTree(
       data.tree.map(hexToBytes),
       data.values,
       data.leafEncoding,
+      hashFn ?? standardHash,
```
Comment (Amxx) [inline src/standard.ts:70]: This is an issue. If the tree was built using a non standard hash, an dumped() to a file, how do you know how to load it? What happens if its loaded using no hashFn, or the wrong hashFn?

When we built this library, we felt like dump/load was essential. Using custom hashes breaks some assumption about the ability to dump/load seemlesly.

**Comment by Amxx** [comment]
Also, I'm don't really like the idea that the StandardMerkleTree, which enforces the use of double hashing, is the place where we support custom hashes.

I'd much rather put that in the SimpleMerkleTree structure (from #31), where leaf hashes are already left to the user's discression, and where the custom hash would be a use defined `bytes32[2] → bytes32` operation. That would not assume the inputs are sorted and concatenated.

File: src/standard.ts (hunk 1 above)
Comment (frangio) [reply src/standard.ts:70]: The dump should contain a line that says `"hash": "custom"`. `load` should have a `hashFn` argument that must be present if the dump contains this line. The user is responsible for passing in the correct hash, although we could do a minimal sanity check at load time to error early (i.e., try to hash something and see that it matches an expected value).

Comment (Amxx) [reply src/standard.ts:70]: Something like this ?
https://github.com/OpenZeppelin/merkle-tree/blob/5e509046ce21d6c34614c5c5d3fee8f0f5492fcb/src/simple.ts#L77-L94

**Comment by ernestognw** [comment]
Closing in favor of #39

**Pull request closed without merging**